File: problem3.py
def factorize(n):
	i = 2
	factors = []
	while n != 1:
		if n%i == 0:
			print(n,'\t|',i,' ==> ',int(n/i))
			n = int(n/i)
			factors.append(i)
			i = 2
		else:
			i+= 1
	else:
		print(factors)
		return max(factors)


number = 256
print(factorize(number))

File: problem3.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Problem3
{
    class Program
    {
        static int[] PrimeFactors(long number)
        {
            List<int> factors = new List<int>();
            while (number != 1)
            {
                for (int i = 1; i <= number; i++)
                {
                    if (number % i == 0)
                    {
                        factors.Add(i);
                        number /= i;
                    }
                }
            }
            // First factor is always going to be 1
            factors.Remove(factors[0]);
            return factors.ToArray();
        }

        static void Main(string[] args)
        {
            int[] factors = PrimeFactors(600851475143);
            Console.WriteLine(factors.Max());
            Console.Read();
        }
    }
}
